planemo/commands/cmd_shed_update.py

```diff
@@ -0,0 +1,74 @@
+"""
+"""
+import sys
+
+import click
+
+from planemo.cli import pass_context
+from planemo import options
+from planemo import shed
+from planemo.io import info, error
+
+
+@click.command("shed_update")
+@options.shed_publish_options()
+@options.shed_upload_options()
+@options.shed_skip_upload()
+@options.shed_skip_metadata()
+@pass_context
+def cli(ctx, paths, **kwds):
+    """Update repository in shed from a ``.shed.yml`` file.
+
+    By default this command will update both repository metadata
+    from ``.shed.yml`` and upload new contents from the repository
+    directory.
+
+    ::
+
+        % planemo shed_update
+
+    This will update the main tool shed with the repository defined
+    by a ``.shed.yml`` file in the current working directory. Both
+    the location of the ``.shed.yml`` and the tool shed to upload to
+    can be easily configured. For instance, the following command can
+    be used if ``.shed.yml`` if contained in ``path/to/repo`` and the
+    desire is to update the test tool shed.
+
+    ::
+
+        % planemo shed_update --shed_target testtoolshed path/to/repo
+
+    Another important option is ``--check_diff`` - this will check for
+    differences before uploading new contents to the tool shed. This may
+    important because the tool shed will automatically populate certain
+    attributes in tool shed artifact files (such as ``tool_dependencies.xml``)
+    and this may cause unwanted installable revisions to be created when
+    there are no important changes.
```

Comment:
This could be a little clearer: Does the diff happen after the meta-data update, but before the tar-ball upload?

e.g. Given a previously up to date tool (in sync with the tool shed), I fix a typo in the ``.shed.yml`` description, and run ``planemo shed_update ...``. What happens (with or without ``--check-diff``)?